tfx_bsl/beam/run_inference.py

```diff
@@ -261,32 +373,79 @@ def update_metrics_with_cache(self):
         self._model_byte_size.update(self.model_byte_size_cache)
         self.model_byte_size_cache = None
 
-    def update(self, elements: List[Union[tf.train.Example,
-                                          tf.train.SequenceExample]],
-               latency_micro_secs: int) -> None:
+    # For feature inputs, using serialized example for batch size
+    def update(
+      self, elements: List[Union[str, bytes]], latency_micro_secs: int) -> None:
       self._inference_batch_latency_micro_secs.update(latency_micro_secs)
       self._num_instances.inc(len(elements))
       self._inference_counter.inc(len(elements))
       self._inference_request_batch_size.update(len(elements))
       self._inference_request_batch_byte_size.update(
-          sum(element.ByteSize() for element in elements))
+          sum(len(element) for element in elements))
+
 
-  def __init__(self, inference_spec_type: model_spec_pb2.InferenceSpecType):
+  def __init__(
+    self, inference_spec_type: model_spec_pb2.InferenceSpecType,
+    tensor_adapter_config: Optional[tensor_adapter.TensorAdapterConfig] = None):
     super(_BaseDoFn, self).__init__()
     self._clock = None
+    self.inference_spec_type = inference_spec_type
     self._metrics_collector = self._MetricsCollector(inference_spec_type)
+    self._tensor_adapter_config = tensor_adapter_config
+    self._io_tensor_spec = None   # This value may be None if the model is remote
 
   def setup(self):
     self._clock = _ClockFactory.make_clock()
 
-  def process(
-      self, elements: List[Union[tf.train.Example, tf.train.SequenceExample]]
-  ) -> Iterable[Any]:
+  def _extract_from_recordBatch(self, elements: pa.RecordBatch):
+    """
+    Function to extract the compatible input with model signature
+    """
+    serialized_examples = None
+    for column_name, column_array in zip(elements.schema.names, elements.columns):
+      if column_name == _RECORDBATCH_COLUMN:
+        column_type = column_array.flatten().type
+        if not (pa.types.is_binary(column_type) or pa.types.is_string(column_type)):
+          raise ValueError(
+            'Expected a list of serialized examples in bytes or as a string, got %s' %
+            type(example))
+        serialized_examples = column_array.flatten().to_pylist()
+        break
+
+    if (serialized_examples is None):
+      raise ValueError('Raw examples not found.')
+
+    model_input = None
+    if self._io_tensor_spec is None:    # Case when we are running remote inference
+      prepare_instances_serialized = (
+        self.inference_spec_type.ai_platform_prediction_model_spec.use_serialization_config)
+      model_input = bsl_util.RecordToJSON(elements, prepare_instances_serialized)
```

Comment:
Override _extract_from_recordBatch in RemotePredictDoFn could separate this out.